include/envoy/http/filter.h

```diff
@@ -198,6 +198,16 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    * @param trailers supplies the trailers to encode.
    */
   virtual void encodeTrailers(HeaderMapPtr&& trailers) PURE;
+
+  /**
+   * Called when a decoder filter goes over its high watermark.
```

Comment:
Is it the filter going over the high watermark or the underlying buffers for its stream/connection? This is a distinction that is interesting because at some point the filter itself will have a notion of low/high watermark when managing its own buffers.